### ISSUE TITLE
Mouse cursor strangely moves by itself or the search for the invisible hand

### ISSUE THREAD

**Issue by baimafeima** [solus-project/budgie-desktop#1549]
It happened several times that the mouse cursor strangely moves by itself, usually towards the right side of the screen. I'm on everything latest (stable repo) running Solus Budgie. 

Hardware: Thinkpad X220

**Comment by baimafeima** [comment]
It was also reported here: https://www.reddit.com/r/SolusProject/comments/9e8v3f/mouse_drift/
In my case, I use a trackpoint on a regular Lenovo laptop.

**Comment by KloudKoder** [comment]
I found that, when this happens, I can hit Super+L to lock the screen. Then I just enter my password in order to log back in, and the mouse is normal again. No doubt something in this code path causes a reset of its state. I've seen many variants of this same bug across many operating systems over years and years, and it's about time we had a "mouse reset" key combo that we could hit in order to fix it. The problem is that sometimes these issues are really hard to fix because a hardware notification gets missed by the OS during periods of high traffic, or when the CPU is doing uninterruptible security stuff. In this case, though, perhaps it's a more hopeful situation because it seems to happen with my USB mouse as well as my touchpad, suggesting that it's due to corruption of the software stack as opposed to a failure to keep synced with some particular mouse hardware.

This issue reminds me of the old keyboard bugs where a shift state keystroke would get missed, causing the machine to behave as though Shift (or Ctrl etc.) were constantly held down, while in fact it had been released.
